pkg/generate/code/set_resource.go

```diff
@@ -955,18 +983,33 @@ func SetResourceIdentifiers(
 			// throwing an error accessible to the user
 			additionalKeyOut += fmt.Sprintf("%s%s, %sok := %s\n", indent, fieldIndexName, fieldIndexName, sourceAdaptedVarName)
 			additionalKeyOut += fmt.Sprintf("%sif %sok {\n", indent, fieldIndexName)
-			additionalKeyOut += fmt.Sprintf("%s\t%s%s.%s = &%s\n", indent, targetVarName, memberPath, cleanMemberName, fieldIndexName)
+
+			switch sourceMemberShape.Type {
+			case "list", "structure", "map":
+				// TODO(RedbackThomson): Add support for slices and maps
+				// in ReadMany operations
+				break
+			default:
+				additionalKeyOut += setResourceForScalar(
+					cfg, r,
+					targetField.Path,
+					targetVarPath,
+					fmt.Sprintf("&%s", fieldIndexName),
+					memberShapeRef,
+					indentLevel+1,
+				)
+			}
 			additionalKeyOut += fmt.Sprintf("%s}\n", indent)
 
 			additionalKeyCount++
 		}
 	}
 
 	// Only use at most one of ARN or nameOrID as primary identifier outputs
-	if arnOut != "" {
+	if primaryIdentifier == "ARN" || primaryKeyOut == "" {
 		return arnOut + additionalKeyOut
 	}
-	return primaryKeyOut + additionalKeyOut
+	return primaryKeyConditionalOut + primaryKeyOut + additionalKeyOut
```

Comment:
These code changes are perfectly fine. I do think you might want to break the function into a couple smaller helper functions in a future PR, though. The function is still pretty longish. :)

Reply:
Yeah this method has grown and grown over time. Will break it out when I implement https://github.com/aws-controllers-k8s/community/issues/874